sql/core/src/test/scala/org/apache/spark/sql/streaming/FlatMapGroupsWithStateSuite.scala

```diff
@@ -1306,22 +1316,22 @@ class FlatMapGroupsWithStateSuite extends StateStoreMetricsTest {
               Update, GroupStateTimeout.NoTimeout, it)(flatMapGroupsWithStateFunc)
 
         testStream(result, Update)(
-          AddData(inputData, "keyOnlyInData", "keyInStateAndData-1"),
+          AddData(inputData, "keyOnlyInData", "keyInStateAndData-2"),
           CheckNewAnswer(
-            ("keyOnlyInState-1", Seq[String](), "2"),
-            ("keyOnlyInState-2", Seq[String](), "1"),
-            ("keyInStateAndData-1", Seq[String]("keyInStateAndData-1"), "2"), // inc by 1
-            ("keyInStateAndData-2", Seq[String](), "1"),
+            ("keyOnlyInState-1", Seq[String](), "1"),
+            ("keyOnlyInState-2", Seq[String](), "2"),
+            ("keyInStateAndData-2", Seq[String]("keyInStateAndData-2"), "3"), // inc by 1
+            ("keyInStateAndData-1", Seq[String](), "1"),
```

Comment:
`keyInStateAndData-1` is NOT in data. this is confusing!

Reply:
hmm it is added in second batch

Reply:
this is nit, the point naming it like this is its in both state and data of the first batch which is what we are mainly testing. hence it is confusing.